recipes-core/rcu-sanity-check/files/post-update-sanity-check

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+rm /data/rcu-service/firmware/update-staged
+rm /etc/rcu-mender-image-firstboot
+
+RCU_SERVICE_STATUS=$(systemctl is-active rcu-service)
+ETH0_STATUS=$(cat /sys/class/net/eth0/operstate)
+
+if [ "$RCU_SERVICE_STATUS" = "active" ]; then
+    mender commit
+    touch /data/rcu-service/firmware/update-succeeded
+    systemctl restart rcu-service
```

Comment:
Why a restart is required?

The other question: why not let rcu-service itself do the checking and commit? We can make the rcu-service to check for the flag after it reaches operational state and commit. That ensures rcu-service is in good shape to serve.

Reply:
I think the concern is that rcu-service may fail before reaching the checking part

Reply:
@kengsooncheah @shlooiNI This does remind me of one issue with this approach..

The original reason we wanted a script to do the sanity check instead of rcu-service was so that if rcu-service is dead, we could still proceed to sanity check other things like image version, available interfaces, etc. The separate script will also be able to flag the update as failed (since rcu-service cannot do so). However, the current approach with the script causes some circular dependency- We want the script to run after rcu-service starts (so that it can detect rcu-service is active), but it also needs to set the 'update-succeeded' flag before rcu-service consumes the flag. The restart there was to make sure that rcu-service picks up the flag and reports RCU last update status correctly.

Having rcu-service perform the sanity check can eliminate these problems. I'll need to change some logic with how rcu-service interprets the 'update-staged' flag. Combining with the /etc/rcu-mender-image-firstboot, I think we can make rcu-service differentiate between having to do sanity check and flagging the last update as a failed update.

Reply:
In short, I think we can go with the approach of getting rcu-service to do the sanity check- We'll still need a recipe which installs the first-boot flag. Combined with the 'update-staged' flag, rcu-service will be able to do the following, depending on which flags are present when RCU (re)boots:

'first-boot' flag present but 'update-staged' flag not present
 - TEZI installation, or mender update which did not go through gRPC API.
 - No sanity check is done, RCU last update status is unknown

'first-boot' flag present and 'update-staged' flag present
- Mender update through gRPC API
- rcu-service will perform sanity check and mender commit if everything goes fine
- If sanity check fails, rcu-service tries to set 'update-failed' flag (and rollback)

'first-boot' flag not present and 'update-staged' present
- Mender update through gRPC API occurred, but rcu-service or image failed to run in previous boot.
- rcu-service should flag 'update-failed'

Reply:
@kengsooncheah @shlooiNI If the above looks good, I'll close this PR and open another PR for a recipe which only installs the first-boot flag.

Reply:
When will rcu-service perform the sanity check? If it performs it too early, it may not detect potential issue. But then if we try to leave the sanity check to later, rcu-service may have crashed, and the sanity check will not be run.

Reply:
@shlooiNI My plan is we do sanity check after rcu-service enters operational state. If rcu-service crashes and the sanity check fails to run, we'd consider the sanity check to have failed.

Reply:
In the case where rcu-service crashes, how would mender rollback be called?

Reply:
@shlooiNI In that case, the user will need to reboot the RCU. Without mender commit being executed by rcu-service, we expect the RCU to boot back to the old partition. In this case, the bootloader handles the rollback for us. For us, calling `mender rollback` explicitly prevents `mender commit` from being called in future, before the RCU reboots to the old image.

Reply:
ok, so means that user will need to physically access and reboot the rack, since remote access may not be available.

Reply:
@HatsyRei, am I right that there are 3 files/flags that are used for communicating the status/state of the firmware update? They are 
1. /etc/rcu-mender-image-firstboot
2. /data/rcu-service/firmware/update-succeeded
3. /data/rcu-service/firmware/update-staged
Any that I missed?

Reply:
@kengsooncheah One more- /data/rcu-service/firmware/update-failed

Reply:
@HatsyRei it looks like these flags are mutually exclusive (no meaning if 2 of them appear a the same time), I wonder if we can collapse them into a single flag? say rcu-update-status and the content could be (installing, staged, verifying, succeeded, failed). I tried to figure out the overall process and drawn it here https://lucid.app/lucidchart/57148ac9-a38b-46a8-8b2a-fb878542d7f7/edit?viewport_loc=-11%2C-23%2C2219%2C1100%2C0_0&invitationId=inv_0fa1682b-2256-4df2-a3e5-d6691af1c2b3 for me to better understand the flow. Will talk to you offline.

Reply:
Since we are going to change the approach to track the firmware update stages and need to rewrite this PR. I'm going to stop reviewing it. One thing to keep in mind is to make the design simple so that it's easy to debug.